test/qa/test_connect.py

```diff
@@ -468,4 +468,73 @@ def test_connect_to_unavailable_servers(tech, proto, obfuscated):
             sh.nordvpn.connect(name)
 
         print(ex.value)
-        assert lib.is_connect_unsuccessful(ex)
\ No newline at end of file
+        assert lib.is_connect_unsuccessful(ex)
+
+
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_status_disconnected(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    assert network.is_disconnected()
+    assert "Disconnected" in sh.nordvpn.status()
+
+
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(60)
+def test_status_connected(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    with lib.Defer(sh.nordvpn.disconnect):
+        name, hostname = server.get_hostname_by(technology=tech, protocol=proto, obfuscated=obfuscated)
+        sh.nordvpn.connect(hostname.split(".")[0])
+
+        connect_time = int(time.time())
+
+        data_10MB = requests.get("http://speedtest.tele2.net/10MB.zip")
+        destination_file = "file_10MB"
+        if data_10MB.status_code == 200:
+            with open(destination_file, "wb") as file:
```

Comment:
1. Instead of opening the same file two times you could just continue the uploading in this same `with` block. Actually I think you also forgot to close the file after opening it the second time below. By the way that's why this `with` construct is used - can't forget to close the file because it is automatically closed when the `with` block ends.
2. It's likely the case that this file could pollute our codespace: after running tests this file would appear as newly added in git, maybe someone would accidentally commit it, etc. Alternative would be to use an in-memory file, so a physical one would not appear https://stackoverflow.com/a/44672691/10169568